transform/instance.go

```diff
@@ -369,20 +369,24 @@ func (ot *objectTransformer) objectTransformJSON(in interface{}, modifier pathMo
 
 // objectTransformXML retrieves the value for this object by building the value for the base object and then adding in any
 // transforms for all defined child fields. If a transform is provided it transforms the children relative to the
-// passed in node
+// passed in node. If a transform is provided and not found the children of the object are skipped.
 func (ot *objectTransformer) objectTransformXML(in interface{}, modifier pathModifier) (interface{}, error) {
 	path := ot.jsonPath
 	if modifier != nil {
 		path = modifier(path)
 	}
 
-	// For the object use a transform if it exists or the default or an empty map
+	// For the object use a transform if it exists, if the transform does not find a node it will return nil
 	if ot.transforms != nil {
 		rawValue, err := ot.transforms.transform(in, "object", modifier, ot.format)
 		if err != nil {
 			return nil, err
 		}
 
+		if rawValue == nil {
```

Comment:
Looks good, my only thought is what happens if the transform comes up empty but a default value is set. I think on the JSON side in such a case the default value is used for the object value and then the children transformations proceed.